core/local/chokidar/send_to_prep.js

```diff
@@ -0,0 +1,186 @@
+  prep /*: Prep */
+) => {
+  const logError = err => log.error({ err, path: folderPath })
+  const doc = metadata.buildDir(folderPath, stats, old.remote)
+  // $FlowFixMe we set doc.overwrite to true, it will be replaced by metadata in merge
+  if (overwrite) doc.overwrite = overwrite
+  log.info({ path: folderPath, oldpath: old.path }, 'DirMove')
+  return prep.moveFolderAsync(SIDE, doc, old).catch(logError)
+}
+
+// New directory detected
+const onAddDir = (
+  { path: folderPath, stats } /*: LocalDirAddition */,
+  prep /*: Prep */
+) => {
+  const doc = metadata.buildDir(folderPath, stats)
+  log.info({ path: folderPath }, 'DirAddition')
+  return prep
+    .putFolderAsync(SIDE, doc)
+    .catch(err => log.error({ err, path: folderPath }))
+}
+
+// File deletion detected
+//
+// It can be a file moved out. So, we wait a bit to see if a file with the
+// same checksum is added and, if not, we declare this file as deleted.
+const onUnlinkFile = (
+  { path: filePath } /*: LocalFileDeletion */,
+  prep /*: Prep */
+) => {
+  log.info({ path: filePath }, 'FileDeletion')
+  return prep
+    .trashFileAsync(SIDE, { path: filePath })
+    .catch(err => log.error({ err, path: filePath }))
+}
+
+// Folder deletion detected
+//
+// We don't want to delete a folder before files inside it. So we wait a bit
+// after chokidar event to declare the folder as deleted.
+const onUnlinkDir = (
+  { path: folderPath } /*: LocalDirDeletion */,
+  prep /*: Prep */
+) => {
+  log.info({ path: folderPath }, 'DirDeletion')
+  return prep
+    .trashFolderAsync(SIDE, { path: folderPath })
+    .catch(err => log.error({ err, path: folderPath }))
+}
+
+// File update detected
+const onChange = (
+  {
+    path: filePath,
+    stats,
+    md5sum
+  } /*: LocalFileUpdate|LocalFileAdded|LocalFileUpdated */,
+  prep /*: Prep */
+) => {
+  log.info({ path: filePath }, 'FileUpdate')
+  const doc = metadata.buildFile(filePath, stats, md5sum)
+  return prep.updateFileAsync(SIDE, doc)
+}
+
+// @TODO inline this.onXXX in this function
+// @TODO rename LocalChange types to prep.xxxxxx
+const step = async (
+  changes /*: LocalChange[] */,
+  { pouch, prep } /*: SendToPrepOpts */
+) => {
+  const errors /*: Error[] */ = []
+  for (let c of changes) {
+    try {
+      if (c.needRefetch) {
+        // $FlowFixMe
+        c.old = await pouch.db.get(metadata.id(c.old.path))
+      }
+      switch (c.type) {
+        // TODO: Inline old LocalWatcher methods
+        case 'DirDeletion':
+          await onUnlinkDir(c, prep)
+          break
+        case 'FileDeletion':
+          await onUnlinkFile(c, prep)
+          break
+        case 'DirAddition':
+          await onAddDir(c, prep)
+          break
+        case 'FileUpdate':
+          await onChange(c, prep)
+          break
+        case 'FileAddition':
+          await onAddFile(c, prep)
+          break
+        case 'FileMove':
+          await onMoveFile(c, prep)
+          if (c.update) await onChange(c.update, prep)
```

Comment:
@taratatach Yesterday bug was here: I was passing `c` instead of `c.update` 🤦‍♂

Reply:
Good catch tests! :+1: